File: ruby/217-Contains-Duplicate.rb
# @param {Integer[]} nums
# @return {Boolean}
def contains_duplicate(nums)
  nums_hash = Hash.new
  nums.each { |value|
    case
    when nums_hash[value] != nil; return true
    else nums_hash[value] = value
    end
  }
  return false
end

File: go/blind_75/arrays_&_hashing/217-Contains-Duplicate.go
package arrayAndHashing

func containsDuplicate(nums []int) bool {
	freqMap := make(map[int]int)

	for i := 0; i < len(nums); i++ {
		if _, ok := freqMap[nums[i]]; ok {
			return true
		}
		freqMap[nums[i]] += 1
	}
	return false
}
